src/app/showcase/service/nodeservice.ts

```diff
@@ -763,6 +764,31 @@ export class NodeService {
         ];
     }
 
+    getDynamicTreeNodes(parentCount: number, childrenCount: number): TreeNode[] {
```

Comment:
Thanks for your contribution.
I would like to suggest this code. Similar to your code but maybe a little bit more clear IMO.
If you add it, test it first because I'm writting without try it. :)

```
getDynamicTreeNodes(parentCount: number, childrenCount: number): TreeNode[] {
    const nodes: TreeNode[] = [];

    for (let index = 0; index < parentCount; index++) {
        const children: TreeNode[] = [];

        for (let j = 0; j < childrenCount; j++) {
            children.push({
                key: `${index}-${j}`,
                label: `Child ${index}-${j}`,
                selectable: true,
            });
        }

        nodes.push({
            key: index.toString(),
            label: `Parent ${index}`,
            selectable: true,
            children: children,
        });
    }

    return nodes;
}
```